modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -361,33 +348,21 @@ public void registerSortedIndex(
     public void unregisterIndex(UUID indexId) {
         indexLockerFactories.remove(indexId);
         indexStorageAdapterFactories.remove(indexId);
+
+        CompletableFuture<?> indexToWaitFuture = indexesToWait.remove(indexId);
+
+        if (indexToWaitFuture != null) {
+            indexToWaitFuture.complete(null);
+        }
     }
 
     private void awaitIndexes() {
         // TODO: replace with actual call to ids supplier
-        List<UUID> indexIds = List.of(pkId()); // activeIndexIds.get();
-
         List<CompletableFuture<?>> toWait = new ArrayList<>();
```

Comment:
Just a nitpick, but you can use an array directly, without an intermediate `ArrayList`

Reply:
Just an array will not work for me, since the size of a `ConcurrentHashMap` can change at any time.

Reply:
ok, makes sense